libs/wingsdk/src/target-tf-azure/bucket.ts

```diff
@@ -0,0 +1,83 @@
+import { ResourceGroup } from "@cdktf/provider-azurerm/lib/resource-group";
+import { StorageAccount } from "@cdktf/provider-azurerm/lib/storage-account";
+import { StorageContainer } from "@cdktf/provider-azurerm/lib/storage-container";
+import { Construct } from "constructs";
+import * as cloud from "../cloud";
+import * as core from "../core";
+import { App } from "./app";
+
+/**
+ * Azure implementation of `cloud.Bucket`.
+ *
+ * @inflight `@winglang/wingsdk.cloud.IBucketClient`
+ */
+export class Bucket extends cloud.BucketBase {
+  private readonly bucket: StorageContainer;
+  private readonly storageAccount: StorageAccount;
+  private readonly resourceGroup: ResourceGroup;
+  private readonly public: boolean;
+
+  constructor(scope: Construct, id: string, props: cloud.BucketProps) {
+    super(scope, id, props);
+
+    this.public = props.public ?? false;
+
+    const app = App.of(this);
+
+    if (app === undefined) {
+      throw new Error("Unable to find app in scope");
+    }
```

Comment:
`App.of` should throw if an app can't be found instead of returning undefined (fail fast by default). Users can always try/catch in the exceptional case that they want to do something if it fails